Add multidrop support to LAN867x tcp transfer example (#27)
espressif/esp-eth-drivers: lan867x/README.md

esp_eth_phy_t *phy = esp_eth_phy_new_lan867x(&phy_config);
```

#### Collision avoidance methods: CSMA/CD and PLCA

> In most cases you can rely on CSMA/CD, which is enabled by default, to avoid collisions when connecting multiple nodes. Its benefits are that it requires zero configuration to work, and does a good job of coordinating when data is sent, however, due to its nature, the delays between transmissions are random, and in timing-critical applications, you are better with deterministic PLCA (PHY-Level Collision Avoidance). Read more about PLCA and its usage on [ieee802.org](https://www.ieee802.org/3/cg/public/July2018/PLCA%20FAQ.pdf).

Reviewer: Consider

```suggestion
In most cases you can rely simply on CSMA/CD (Carrier Sense Multiple Access / Collision Detection) provided by MAC layer to avoid collisions when connecting multiple nodes. It requires zero configuration to work, and does a good job of coordinating when data is sent. However, due to its nature, the delays between transmissions are random. For timing-critical applications, deterministic PLCA (PHY-Level Collision Avoidance) provides a guaranteed maximum latency along with improved throughput. Read more about PLCA and its usage at [ieee802.org](https://www.ieee802.org/3/cg/public/July2018/PLCA%20FAQ.pdf).
```

Author: Applied some changes, with a slight rewording of guaranteed maximum latency.

> In most cases you can rely on the default CSMA/CD (Carrier Sense Multiple Access / Collision Detection) provided by the MAC layer, to avoid collisions when connecting multiple nodes. It requires zero configuration to work and does a good job of coordinating when data is sent. However, due to its nature, the delays between transmissions are random. For timing-critical applications, deterministic PLCA (PHY-Level Collision Avoidance) provides a guaranteed time period between transmissions along with improved throughput. Read more about PLCA and its usage at [ieee802.org](https://www.ieee802.org/3/cg/public/July2018/PLCA%20FAQ.pdf).

Reviewer: ```suggestion
In most cases you can rely on the default CSMA/CD (Carrier Sense Multiple Access / Collision Detection) provided by the MAC layer to avoid collisions when connecting multiple nodes. It requires zero configuration to work and does a good job of coordinating when data is sent. However, due to its nature, the delays between transmissions are random. For timing-critical applications, deterministic PLCA (PHY-Level Collision Avoidance) provides a guaranteed time period between transmissions along with improved throughput. Read more about PLCA and its usage at [ieee802.org](https://www.ieee802.org/3/cg/public/July2018/PLCA%20FAQ.pdf).
```

| LAN867X_ETH_CMD_RM_TX_OPPORTUNITY  | uint8_t* node_id | Remove additional transmit opportunity for chosen node                                        |
| LAN867X_ETH_CMD_S_MAX_BURST_COUNT  | uint8_t* maxcnt  | Set max count of additonal packets transmitted during one frame, or 0 to disable PLCA burst   |
| LAN867X_ETH_CMD_G_MAX_BURST_COUNT  | uint8_t* maxcnt  | Write max count of additonal packets transmitted during one frame to the location via pointer |
| LAN867X_ETH_CMD_S_BURST_TIMER      | uint8_t* time    | Set time during which additional packets in BTs                                               |
| LAN867X_ETH_CMD_G_BURST_TIMER      | uint8_t* time    | Write time during which additional packets can be sent in BTs to the location via pointer     |
| LAN768X_ETH_CMD_PLCA_RST           |                  | Perform reset of the PLCA                                                                     |

One of the devices on the network must be a **coordinator** for which you need to set _node count_ to amount of connected nodes, and _ID_ to 0.

> On all other nodes, only _ID_ is required, which must be unique for every node

Reviewer: ```suggestion
On all other nodes, only _ID_ is required, which must be unique for every node.
```

#### You may need to setup the PLCA

> Usually you can rely on CSMA/CD, which is enabled by default, to avoid collisions when connecting multiple nodes. However in some cases PLCA may need to be used if you are still getting collisions.

Author: Added a paragraph on CSMA/CD and a paragraph on PLCA.

| LAN867X_ETH_CMD_RM_TX_OPPORTUNITY  | uint8_t* node_id | Remove additional transmit opportunity for chosen node                                        |
| LAN867X_ETH_CMD_S_MAX_BURST_COUNT  | uint8_t* maxcnt  | Set max count of additonal packets transmitted during one frame, or 0 to disable PLCA burst   |
| LAN867X_ETH_CMD_G_MAX_BURST_COUNT  | uint8_t* maxcnt  | Write max count of additonal packets transmitted during one frame to the location via pointer |
| LAN867X_ETH_CMD_S_BURST_TIMER      | uint8_t* time    | Set time during which additional packets in BTs                                               |
| LAN867X_ETH_CMD_G_BURST_TIMER      | uint8_t* time    | Write time during which additional packets can be sent in BTs to the location via pointer     |
| LAN768X_ETH_CMD_PLCA_RST           |                  | Perform reset of the PLCA                                                                     |

One of the devices on the network must be a **coordinator** for which you need to set _node count_ to amount of connected nodes, and _ID_ to 0.
On all other nodes only _ID_ is required, which must be unique for every node

> After that he device is ready and the Ethernet driver can be used as normal. For more information of how to use ESP-IDF Ethernet driver, visit [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/network/esp_eth.html).

Reviewer: Check for formatting and typos.

> #### You may need to setup the PLCA

Reviewer 2: This section could have a better title.

Author: Will be addressed in the next commit along with other documentation changes.

In most cases you can rely on CSMA/CD, which is enabled by default, to avoid collisions when connecting multiple nodes. Its benefits are that it requires zero configuration to work, and does a good job of coordinating when data is sent, however, due to its nature, the delays between transmissions are random, and in timing-critical applications, you are better with deterministic PLCA. 

> PLCA requires setting at least PLCA ID, node count, and selecting one node as a coordinator but ensures precise and repeatable time between transmissions, and due to the flexibility of the transmit opportunity timer, burst mode and additional transmit opportunities you can fine-tune it to your needs.

Reviewer 2: We could add links to both CSMA/CD and PLCA

Reviewer: I think providing link (or other type of reference) to LAN867x how to configure PLCA will be sufficient. No link needed for CSMA/CD since there is no configuration required and it's kind of old approach so there is bunch of information about it on the internet.
